dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStoreDir.java

```diff
@@ -114,36 +114,56 @@ private Optional<PageInfo> getPageInfo(Path path) {
         createdTime = creationTime.toMillis();
       } catch (IOException e) {
         LOG.error("Failed to get file size for " + path, e);
+        deleteUnrecognizedPage(path);
         return Optional.empty();
       }
       return Optional.of(new PageInfo(pageId.get(),
           pageSize, CacheScope.GLOBAL, this, createdTime));
     }
+    deleteUnrecognizedPage(path);
     return Optional.empty();
   }
 
+  /**
+   * Deletes an unrecognized page file due to various reasons.
+   * @param path the file path of a page file
+   */
+  private void deleteUnrecognizedPage(Path path) {
+    Preconditions.checkState(path.startsWith(getRootPath()),
+        String.format("%s is not inside the cache dir (%s)!", path, getRootPath()));
+    try {
+      Files.delete(path);
+    } catch (IOException e) {
+      // ignore.
+    }
+  }
+
   /**
    * @param path path of a file
-   * @return the corresponding page id, or null if the file name does not match the pattern
+   * @return the corresponding page id, or empty if the file name does not match the pattern
    */
   private Optional<PageId> getPageId(Path path) {
     Matcher matcher = mPagePattern.matcher(path.toString());
     if (!matcher.matches()) {
-      LOG.error("Unrecognized page file " + path);
+      // Please note, TEMP page files are also treated as Unrecognized!!
+      LOG.error("Unrecognized page file " + path + ". Let's delete it.");
```

Comment:
can we silence the log for temp pages? only print an error in case of a truly unrecognized page? because the entire temp page directory should be cleared after a restart, they are expected and not quite an error.

Reply:
OK.

Reply:
Fixed.